### PR TITLE
Update API Typings for Better and More Easily Accessible Code Completion

### PR DESCRIPTION
**Pull request by ErrorGamer2000**
### Changes

Added `.d.ts` files for more accurate types, and added types to existing addons.

### Reason for changes

The current typings are not really useful when editing.

### Tests

Tested with intellisense in VSCode.

### Jobs done

- [x] Add typings
- [ ] Update docs with a "How to enable intellisense and code completion" section for both userscripts and popups.

### CODE REVIEW

**Comment by ErrorGamer2000** [comment]
@WorldLanguages will you please check this?

**Comment by WorldLanguages** [comment]
@ErrorGamer2000 I DMed you on Discord, sorry I didn't know you weren't checking it

**Comment by ErrorGamer2000** [comment]
Oh, yeah. Sorry about that. I am at Disney world so can't really check because no laptop and parents constantly with me. Leave tomorrow afternoon and get home around midnight, so maybe the day after. Sorry.

File: addon-api/types/common.d.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  /**
+   * Fetch whether the user is logged in or not
+   * @returns whether the user is logged in or not.
+   */
+  fetchIsLoggedIn(): Promise<boolean>;
+  /**
+   * Fetch current username.
+   * @returns the username.
+   */
+  fetchUserName(): Promise<string | void>;
```
Comment (Weredime) [inline addon-api/types/common.d.ts:66]:
```suggestion
  fetchUsername(): Promise<string | void>;
```

File: addon-api/types/content-script.d.ts [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+     */
+    get clientVersion(): "scratch-www" | "scratchr2" | null;
+    /**
+     * Add a custom block to the user's workspace.
+     * Do not use this unless you are adding blocks to the debugger addon.
+     * @param proccode The name that will be used to refer to the block internally.
+     * Must specify inputs on the block (if any) using %s (string), %n (number-only), and %b (boolean).
+     * @param options Block options.
+     */
+    addblock<A extends string[]>(
```
Comment (Weredime) [inline addon-api/types/content-script.d.ts:111]:
```suggestion
    addBlock<A extends string[]>(
```

**Pull request closed without merging**